packages/build-utils/src/get-project-paths.ts

```diff
@@ -0,0 +1,67 @@
+import { detectFramework } from './detect-framework';
+import { DetectorFilesystem } from './detectors/filesystem';
+import frameworks from '@vercel/frameworks';
+
+const MAX_DEPTH_TRAVERSE = 3;
+
+export interface GetProjectPathsOptions {
+  fs: DetectorFilesystem;
+  path?: string;
+  skipPaths?: string[];
+}
+
+export type ProjectPath = string;
+
+export const getProjectPaths = async ({
+  fs,
+  path,
+  skipPaths,
+}: GetProjectPathsOptions): Promise<ProjectPath[] | any> => {
+  const allPaths: Array<ProjectPath> = [];
+  for (
+    let currentDepth = 0;
+    currentDepth < MAX_DEPTH_TRAVERSE;
+    currentDepth++
+  ) {
+    if (path && skipPaths?.includes(path)) {
+      return allPaths;
+    } else {
+      const directoryContents = await fs.readdir(path ?? './');
+
+      // This should only ever return one file bc there should never be more than one package.json in a directory
+      const childFiles = directoryContents.filter(
```

Comment:
```suggestion
      const hasPackageJson = directoryContents.some(
```

You can use a boolean instead and it will exit early instead of traversing all files.